src/plugins/vis_builder/public/application/components/top_nav.tsx

```diff
@@ -34,6 +36,10 @@ export const TopNav = () => {
 
   const saveDisabledReason = useCanSave();
   const savedVisBuilderVis = useSavedVisBuilderVis(visualizationIdFromUrl);
+  connectStorageToQueryState(services.data.query, services.osdUrlStateStorage, {
```

Comment:
What happens in the case where `connectStorageToQueryState` fails for some reason? It doesn't look like we have any explicit error handling. Does visbuilder still work as expected without the query state connection?

Reply:
Thanks! I added a simple try catch wrap to the function. I also tested if connectStorageToQueryState fails for some reason, the persistence won't work, but vis builder still work as expected.